File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.ts
// Last updated: 4/8/2026, 2:08:02 AM
function minSubArrayLen(target: number, nums: number[]): number {   
    let left = 0;
    let right = 0;
    let sumOfCurrentWindow = 0;
    let res = Number.MAX_VALUE;

    for (right = 0; right < nums.length; right++) {
        sumOfCurrentWindow += nums[right]

        while (sumOfCurrentWindow >= target) {
            res = Math.min(res, right - left + 1)
            sumOfCurrentWindow -= nums[left++] 
        }
    }
    
    return res === Number.MAX_VALUE ? 0 : res
};

File: 209-MinimumSizeSubarraySum/209-MinimumSizeSubarraySum.kt
// Last updated: 4/8/2026, 2:08:09 AM
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        var n: Int = nums.size
        var ans: Int = Int.MAX_VALUE
        var left: Int = 0
        var sum: Int = 0
        for (i in 0 until n) {
            sum += nums[i]
            while (sum >= target) {
                ans = min(ans, i+1-left)
                sum -= nums[left++]
            }
        }
        if (ans != Int.MAX_VALUE) {
            return ans
        } 
        return 0
    }
}